sources/log_exp_math.move

```diff
@@ -238,4 +245,48 @@ module fixed_point64::log_exp_math {
         let y_times_ln_x = fixed_point64::mul_fp(y, ln_x);
         exp(sign, y_times_ln_x)
     }
+
+    /// Returns square root of x, precisely floor(sqrt(x))
+    /// Adapted from math128: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/math128.move#L143
+    public fun sqrt(x: u256): u256 {
+        if (x == 0) return 0;
+        // Note the plus 1 in the expression. Let n = floor_lg2(x) we have x in [2^n, 2^{n+1}) and thus the answer in
+        // the half-open interval [2^(n/2), 2^{(n+1)/2}). For even n we can write this as [2^(n/2), sqrt(2) 2^{n/2})
+        // for odd n [2^((n+1)/2)/sqrt(2), 2^((n+1)/2). For even n the left end point is integer for odd the right
+        // end point is integer. If we choose as our first approximation the integer end point we have as maximum
+        // relative error either (sqrt(2) - 1) or (1 - 1/sqrt(2)) both are smaller then 1/2.
+        let res = 1 << ((floor_log2(x) + 1) >> 1);
+        // We use standard newton-rhapson iteration to improve the initial approximation.
+        // The error term evolves as delta_i+1 = delta_i^2 / 2 (quadratic convergence).
+        // It turns out that after 5 iterations the delta is smaller than 2^-64 and thus below the treshold.
+        res = (res + x / res) >> 1;
+        res = (res + x / res) >> 1;
+        res = (res + x / res) >> 1;
+        res = (res + x / res) >> 1;
+        res = (res + x / res) >> 1;
```

Comment:
this is u256 rather than u128 - does "// It turns out that after 5 iterations the delta is smaller than 2^-64 and thus below the treshold." still hold true?